### ISSUE TITLE
New Checkout experience won't load asset files if the plugin is symlinked

### ISSUE THREAD

**Issue by laurelfulford** [woocommerce/woocommerce-gateway-stripe#3185]
**Describe the bug**
When the WooCommerce Stripe Gateway is symlinked and the new checkout experience is enabled, none of the assets (images, CSS, JS) load on the front-end.  

**To Reproduce**
1. Set up a test site with the WooCommerce Stripe Gateway plugin symlinked rather than installed directly in the wp-content/plugins directory.
2. Turn on the new checkout experience.
3. Run through a checkout; note that the Stripe form doesn't load on the Payment Info section. In the console, there are 404 errors for the `upe_` files.
4. Turn on the legacy checkout experience.
5. Repeat step 3; note that the Stripe form loads again.

**Expected behavior**
That the assets would load for the new checkout experience when the plugin is symlinked.

**Screenshots**

![image](https://github.com/woocommerce/woocommerce-gateway-stripe/assets/177561/4d3f7950-7c79-42f9-8abe-46241e318915)

The screenshot is from a test site, but this originally was reported from a live site, and can be recreated when Stripe is and is not in test mode. 

**Environment (please complete the following information):**
 - WordPress Version: 6.5.4
 - WooCommerce Version: 8.9.2
 - Stripe Plugin Version: 8.3.1
 - Browser [e.g. chrome, safari] and Version: All
 - Any other plugins installed: I can recreate with just WooCommerce and the Stripe Gateway installed.

**Comment by claudiulodro** [comment]
![Annotation on 2024-06-06 at 13-06-00](https://github.com/woocommerce/woocommerce-gateway-stripe/assets/7317227/1c88a7b2-c1fc-4caa-8406-a6de62210356)

In terms of immediate impact, this affects all WP Cloud sites that use a symlinked version of the Stripe gateway.

**Comment by annemirasol** [comment]
I am unable to reproduce this, both on the latest versions of WordPress, WooCommerce and Stripe plugin (6.6.1, 9.2.3, 8.6.1, respectively) and on the reported versions (6.5.4, 8.9.2, 8.3.1)

<img width="1564" alt="Screenshot 2024-09-04 at 2 21 18 PM" src="https://github.com/user-attachments/assets/035d3c40-1dbe-43c5-b352-fb6b126a5739">

@laurelfulford @claudiulodro Thank you for reporting this. Are you still experiencing this issue, even with the newer versions? Is there anything in my setup (shown above) that's different from yours?

**Comment by annemirasol** [comment]
Closing this as I am unable to reproduce.

**Comment by mjonasftc** [comment]
@annemirasol We had a publisher who saw this same issue on two sites today. I'm not an engineer but let me know if there's any other relevant information you'd like from me.

**Comment by annemirasol** [comment]
@mjonasftc Can you ask what their WordPress, WooCommerce, and Stripe extension versions are?

Also, can they do `ls -la` inside their WordPress `wp-content/plugins/` folder, and paste here the line that shows us the symlink for the plugin? It should look something like this:

```
lrwxr-xr-x@  1 annemirasol  staff    42 Oct 11 09:46 test.php -> /Users/annemirasol/dev/my-plugins/test.php
```

I will re-open this issue.

**Comment by claudiulodro** [comment]
```
woocommerce-gateway-stripe -> ../../../wordpress/plugins/woocommerce-gateway-stripe/latest
```

I've set up a simple example store at https://stripe-gateway-issue.newspackstaging.com/shop/ demonstrating the issue. Feel free to give me a ping in Slack if you need access or have questions about the WP Cloud/Atomic infrastructure.

![Image](https://github.com/user-attachments/assets/ae30cbab-b688-4460-a734-c534039433aa)

**Comment by dkoo** [comment]
I tried to fix this in a [now-stale PR](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/3029) a few months ago, but perhaps it didn't fully address the problem. Linking it here in case it's still useful.

**Comment by annemirasol** [comment]
Thanks, @claudiulodro and @dkoo! I will focus on this issue today, and look at @dkoo's PR.

**Comment by annemirasol** [comment]
Fixed by #3029

**Comment by claudiulodro** [comment]
Thanks for reviewing!
